py/server/deephaven/_udf.py

```diff
@@ -104,10 +105,11 @@ def _parse_param_annotation(annotation: Any) -> _ParsedParamAnnotation:
     return p_annotation
 
 
-def _parse_type_no_nested(annotation: Any, p_annotation: _ParsedParamAnnotation, t: type) -> None:
+def _parse_type_no_nested(annotation: Any, p_annotation: _ParsedParamAnnotation, t: Union[type, str]) -> None:
     """ Parse a specific type (top level or nested in a top-level Union annotation) without handling nested types
     (e.g. a nested Union). The result is stored in the given _ParsedAnnotation object.
     """
+    t = eval(t) if isinstance(t, str) else t
```

Comment:
Is there a reason the eval can't go in the "old" code block of `_parse_signature` below instead of the logic being split across two files?

Does this line deserve a comment if it stays here?

Reply:
"eval' the specific type hints here makes sense because this is where they are only parsed. `_parse_signature` handles the whole signature, it would be quite cluttering to dive into individual typing there.